### PR TITLE
Add sorting by username to user-list

### PR DESCRIPTION
**Pull request by fhlavac**
- added sorting by username to user-list
- enhanced user-list tests with fetching data on sort test
- fixed fetchGroup() params order causing groups tests failure

@karelhala

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@fhlavac these changes should help with debugging:
```diff
diff --git a/src/smart-components/group/add-group/users-list.js b/src/smart-components/group/add-group/users-list.js
index 1828161..55a11d6 100644
--- a/src/smart-components/group/add-group/users-list.js
+++ b/src/smart-components/group/add-group/users-list.js
@@ -29,7 +29,7 @@ const UsersList = ({ users, fetchUsers, isLoading, pagination, selectedUsers, se
   const [ filterValue, setFilterValue ] = useState('');
 
   useEffect(() => {
-    fetchUsers();
+    fetchUsers()
   }, []);
 
   const setCheckedItems = (newSelection) => {
@@ -73,7 +73,7 @@ const mapStateToProps = ({ userReducer: { users, isUserDataLoading }}) => {
 const mapDispatchToProps = dispatch => {
   return {
     fetchUsers: (apiProps = { limit: 10, offset: 0 }) => {
-      dispatch(fetchUsers(mappedProps(apiProps)));
+      return dispatch(fetchUsers(mappedProps(apiProps)));
     },
     addNotification: (...props) => dispatch(addNotification(...props))
   };
diff --git a/src/test/smart-components/user/users.test.js b/src/test/smart-components/user/users.test.js
index 0ee125d..d50bf40 100644
--- a/src/test/smart-components/user/users.test.js
+++ b/src/test/smart-components/user/users.test.js
@@ -43,7 +43,7 @@ describe('<Users />', () => {
     mock.reset();
   });
 
-  it('should render user list correctly', async() => {
+  it.only('should render user list correctly', async() => {
     mock.onGet(`/api/rbac/v1/principals/?limit=10&sort_order=asc`).replyOnce(200, {});
     let wrapper;
     const store = mockStore(initialState);

```

File: src/helpers/group/group-helper.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const groupApi = getGroupApi();
 
 export async function fetchGroups({ limit, offset, name, orderBy }) {
   const [ groups, auth ] = await Promise.all([
-    groupApi.listGroups(limit, offset, name, undefined, undefined, orderBy),
+    groupApi.listGroups(limit, offset, name, undefined, undefined, undefined, undefined, undefined, orderBy),
```
Comment (Hyperkid123) [inline src/helpers/group/group-helper.js]: :scream: I think we might want to stop this argument madness and create some query builder or custom generator which will use an object for params instead. This is getting silly

(nothing you can do about it at the moment, just pointing out)

**Comment by Hyperkid123** [comment]
@fhlavac seems that some tests are falling on API mocks.

File: src/helpers/group/group-helper.js (hunk 1 above)
Comment (karelhala) [reply src/helpers/group/group-helper.js]: @Hyperkid123 agree, I've wrote something for inventory but we'd need some generic way how to map object to params.

**Comment by karelhala** [comment]
@fhlavac looks like you forgot to push package.json and package-lock.json

**Comment by karelhala** [comment]
Build succeeded https://travis-ci.org/github/RedHatInsights/insights-rbac-ui/builds/668760044

**Pull request merged**